test/test_jit.py

```diff
@@ -1257,7 +1278,55 @@ def fn(x, t):
             return x * y * z
 
         inputs = (torch.randn(1), (torch.randn(1), torch.randn(1)))
-        self.checkTrace(fn, inputs)
+        self.checkTrace(fn, inputs, flatten_inputs=False)
```

Comment:
Confusing that `flatten_inputs=False` has been added to a unit test called `test_input_flatten`. What's going on here?

Reply:
yeah, so...

2 things: I should rename the parameter to `flatten_autograd_inputs`. The test is expecting to fail, because it's "enable me when you teach autograd to flatten its inputs", which I want to do in a followup. When that happens, we can remove all that logic, but this was the quick-n-dirty way to get stuff flattened from the caller side.

I'm happy to make it the default, but we will lose the test that autograd is doing it properly. That's all the flag is protecting.